### ISSUE TITLE
Chart flickers when using groupData: true

### ISSUE THREAD

**Issue by hardikmodha** [amcharts/amcharts4#2305]
Hi, I'm trying to use a dynamic data grouping feature with XYChart. Chart flickers when I use `groupCount: true`. What I have noticed is that for some values the issue occurs, for some values issue don't occur. 

Here is the link to codesandbox(es): 

https://codesandbox.io/s/amcharts-issue-minimal-config-hns40. (`groupCount: 7`)

https://codesandbox.io/s/amcharts-issue-minimal-config-hzu5q (`groupCount: 100`)
 
In the example, I've used version `4.9.12` but I'm also able to reproduce the issue on versions lower than `4.9.12`.

**Comment by hardikmodha** [comment]
Friendly ping. @martynasma @amcharts

**Comment by zeroin** [comment]
Hi,

With the `groupCount` set to 7 and this amount of data some strange thing happens - the group interval switches between day and week indefinitely. I am investigating this, meanwhile a solution would be to remove grouping to weekly data by setting groupIntervals without a week to your dateAxis:
```
groupIntervals:[
      { timeUnit: "hour", count: 1 },
      { timeUnit: "day", count: 1 },
      { timeUnit: "month", count: 1 },
      { timeUnit: "year", count: 1 }
    ]
```

Or changing `groupCount` to some other (bigger) number.

**Comment by martynasma** [comment]
I'm afraid we can't find a good safe solution to accommodate your data/grouping settings, that would not affect general functionality.

I hope workaround proposed by @zeroin is sufficient.

**Comment by hardikmodha** [comment]
Hi, Thank you @zeroin for the workaround. It works. But I would like to mention that the same issue was appearing when I had used the `groupCount` ~100 in the past. I'll try to add an example where it gets reproduced for a bigger number of `groupCount`.

**Comment by hardikmodha** [comment]
@martynasma I can understand the situation. It would be great if you can answer my question on how can one decide the optimal number of `groupCount` so that we can avoid similar issues. Is there such a thing?

**Comment by zeroin** [comment]
I will try to take another look at this if you say this also happens with `groupCount = 100`

**Comment by hardikmodha** [comment]
@zeroin I've updated the details with the reproduction of the issue when `groupCount` is set to 100. https://codesandbox.io/s/amcharts-issue-minimal-config-hzu5q

**Comment by zeroin** [comment]
Thanks, this will help me a lot.

**Comment by zeroin** [comment]
Solved the problem in dev version, we will release the update shortly. The reason why this was not reproduced often is because you use 0.5 for `startLocation` and `endLocation`. And I did not cared about it when calculating how many periods fit into the selection, so the period was switching from one to another because of this. I will let you know when an update is released.

**Comment by hardikmodha** [comment]
That's awesome. Thank you! :) :tada:

**Comment by martynasma** [comment]
Fixed in [4.9.14](https://github.com/amcharts/amcharts4/releases/tag/4.9.14).

## [4.9.14] - 2020-04-13

### Added
- Global options setting added `am4core.options.animationsEnabled` (default `true`). Set it to `false` to **disable all animations on all charts**. Can be used to toggle animations on and off dynamically.
- `SliceGrouper` plugin: new setting `limit` added. If set to a number, will ignore `threshold` but will show only first X slices and will group the rest.
- New PDF format option for `Export`: `align`. Available options: `"left"` (default), `"center"`, and `"right"`.
- New `ValueAxis` property: `adjustLabelPrecision` (default: `true`). `true` means that **all** labels on `ValueAxis` (except zero) will maintain the same number of decimals, e.g. `1.0`, `1.5`, `2.0` versus `1`, `1.5`, `2`. Enabling it will ignore number of active decimals set via `numberFormat`.
- `DateFormatter` is now exposing list of various named time units: `months`, `monthsShort`, `weekdays`, and `weekdaysShort`. You can now resolve numeric value to an English name (e.g. `chart.dateFormatter.weekdaysShort[2]`) or to your active locale (`chart.dateFormatter.language.translate(chart.dateFormatter.weekdaysShort[1])`).

### Changed
- New property `adjustLabelPrecision` hsa a default of `true`, which means that all labels will now have same number of decimals on `ValueAxis`. Set `adjustLabelPrecision = false`, to disable and revert behavior to how it was before this update (`1`, `1.5`, `2`).
- `am4core.options.viewportTarget` now accepts array of references if you have multiple scrollable containers holding charts and have `onlyShowOnViewport` enabled.
- Default `DateAxis.periodChangeDateFormats` for `"hour"` changed from `"_date_hour"` to `"_date_day"` so that the format would differ for midnight (hour 00:00).

### Fixed
- Parsing short month names (`"MMM"`) from string-based dates was failing on May.
- When exporting chart to PDF, on narrow charts, sometimes the whole chart would go to the next page.
- Having `null` values in data was causing `maxTooltipDistance == -1` to show multiple tooltips instead of one.
- It was impossible to have individual `innerRadius` for `AxisRendererRadial` and `AxisRendererCircular` - the `RadarChart`'s `innerRadius` was overriding them.
- Removing axis ranges from an axis could result in a JS error.
- In some cases, when `DateAxis` properties `startLocation` and `endLocation` were not defaults (0 and 1), the chart might flicker from one grouped data period to another continously. It was only happening with some particular `groupCount` values and data item counts, though.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
